pkg/applier/inventory/inventory_converter_test.go

```diff
@@ -0,0 +1,647 @@
+							Namespace: "obj2-ns",
+							GroupKind: v1alpha1.GroupKind{
+								Group: "obj2-group",
+								Kind:  "obj2-kind",
+							},
+						},
+					},
+				},
+				status: v1alpha1.ResourceGroupStatus{
+					ObservedGeneration: int64(invGeneration),
+					ResourceStatuses: []v1alpha1.ResourceStatus{
+						{
+							ObjMetadata: v1alpha1.ObjMetadata{
+								Name:      "obj1-name",
+								Namespace: "obj1-ns",
+								GroupKind: v1alpha1.GroupKind{
+									Group: "obj1-group",
+									Kind:  "obj1-kind",
+								},
+							},
+							Status:     v1alpha1.Current,
+							SourceHash: "abc-123",
+							Strategy:   v1alpha1.Apply,
+							Actuation:  v1alpha1.ActuationSucceeded,
+							Reconcile:  v1alpha1.ReconcileSucceeded,
+						},
+						{
+							ObjMetadata: v1alpha1.ObjMetadata{
+								Name:      "obj2-name",
+								Namespace: "obj2-ns",
+								GroupKind: v1alpha1.GroupKind{
+									Group: "obj2-group",
+									Kind:  "obj2-kind",
+								},
+							},
+							Status:     v1alpha1.Current,
+							SourceHash: "abc-123",
+							Strategy:   v1alpha1.Apply,
+							Actuation:  v1alpha1.ActuationSucceeded,
+							Reconcile:  v1alpha1.ReconcileSucceeded,
+						},
+					},
+				},
+			}.build(t),
+			wantObjRefs: object.ObjMetadataSet{
+				{
+					Name:      "obj1-name",
+					Namespace: "obj1-ns",
+					GroupKind: schema.GroupKind{
+						Group: "obj1-group",
+						Kind:  "obj1-kind",
+					},
+				},
+				{
+					Name:      "obj2-name",
+					Namespace: "obj2-ns",
+					GroupKind: schema.GroupKind{
+						Group: "obj2-group",
+						Kind:  "obj2-kind",
+					},
+				},
+			},
+			// Converter does not currently populate status to Inventory
+			wantObjStatuses: nil,
+		},
+	}
+	for name, tc := range testCases {
+		t.Run(name, func(t *testing.T) {
+			ic := NewInventoryConverter(declared.RootScope, "test-sync", metadata.StatusEnabled)
+			inv, err := ic.InventoryFromUnstructured(tc.fromObj)
+			if tc.wantErr {
+				require.Error(t, err)
+				return
+			}
+			require.NoError(t, err)
+			require.Equal(t, invName, inv.GetName())
+			require.Equal(t, invNamespace, inv.GetNamespace())
+			require.Equal(t, invID, inv.GetID().String())
+			require.Equal(t, tc.wantObjRefs, inv.GetObjectRefs())
+			require.Equal(t, tc.wantObjStatuses, inv.GetObjectStatuses())
+		})
+	}
+}
+
+// inventoryFactory is a convenience test helper for constructing a UnstructuredInventory
+type inventoryFactory struct {
+	objRefs     object.ObjMetadataSet
+	objStatuses []actuation.ObjectStatus
+}
+
+func (f inventoryFactory) build(t *testing.T) *inventory.UnstructuredInventory {
+	emptyRG := resourceGroupFactory{}.build(t)
+	ui := inventory.NewUnstructuredInventory(emptyRG)
+	ui.SetObjectRefs(f.objRefs)
+	ui.SetObjectStatuses(f.objStatuses)
+	return ui
+}
+
+func TestInventoryToUnstructured(t *testing.T) {
+
+	testCases := map[string]struct {
+		fromObj *unstructured.Unstructured
+		toInv   *inventory.UnstructuredInventory
+		wantErr bool
```

Comment:
you can do error comparisons with `testerrors.AssertEqual`

Reply:
done